src/token.rs

```diff
@@ -229,4 +299,78 @@ mod tests {
         assert_eq!(contract.balanceOf(::pwasm_std::ext::sender()), 10000.into());
         assert_eq!(contract.balanceOf("0xdb6fd484cfa46eeeb73c71edee823e4812f9e2e1".into()), 0.into());
     }
+
+    test_with_external!(
+        ExternalBuilder::new().build(),
+        approve_should_approve {
+            let mut contract = TokenContractInstance{};
+            let spender: Address = "0xdb6fd484cfa46eeeb73c71edee823e4812f9e2e1".into();
+            contract.constructor(40000.into());
+            contract.approve(spender, 40000.into());
+            assert_eq!(get_external::<ExternalInstance>().logs().len(), 1, "Should be 1 event logged");
+            assert_eq!(get_external::<ExternalInstance>().logs()[0].topics.as_ref(), &[
+                "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925".into(), // hash of the event name
+                "0x0000000000000000000000000000000000000000000000000000000000000000".into(), // sender (owner) address
+                "0x000000000000000000000000db6fd484cfa46eeeb73c71edee823e4812f9e2e1".into()]); // spender address
+            assert_eq!(contract.allowance(::pwasm_std::ext::sender(), spender.clone()), 40000.into());
+        }
+    );
+
+    test_with_external!(
+        ExternalBuilder::new()
+            // Allowance to spend from tokens owner for spender
+            .storage("e428ed52e2e5765d58bbc65412f7c432d6388f27105bbb1ede5c2badfc4b3b6b".into(), U256::from(10000).into())
+            // Total Sypply
+            .storage("0200000000000000000000000000000000000000000000000000000000000000".into(), U256::from(40000).into())
+            // Contract owner
+            .storage("0300000000000000000000000000000000000000000000000000000000000000".into(), H256::from(Address::new()).into())
+            // Balance
+            .storage("0100000000000000000000000000000000000000000000000000000000000000".into(), U256::from(40000).into())
+            .sender("0xdb6fd484cfa46eeeb73c71edee823e4812f9e2e1".into())
+            .build(),
+        spender_should_be_able_to_spend_if_allowed {
+            let mut contract = TokenContractInstance{};
+            contract.constructor(40000.into());
+            let owner: Address = Address::new();
+            let samAddress: Address = "0xea674fdde714fd979de3edf0f56aa9716b898ec8".into();
+
+            assert_eq!(contract.transferFrom(owner.clone(), samAddress.clone(), 5000.into()), true);
+            assert_eq!(contract.balanceOf(samAddress.clone()), 5000.into());
+            assert_eq!(contract.balanceOf(owner.clone()), 35000.into());
+
+            assert_eq!(contract.transferFrom(owner.clone(), samAddress.clone(), 5000.into()), true);
+            assert_eq!(contract.balanceOf(samAddress.clone()), 10000.into());
+            assert_eq!(contract.balanceOf(owner.clone()), 30000.into());
+
+            // limit reached. No more coins should be available to spent for the spender
```

Comment:
I'm bad at english, but isn't this should be "for spending" instead of "to spent"?

Reply:
or at least "to spend"